### DISCUSSION TITLE
CLCapture 'capture' mode returns all zeros. All 3 UF2s, Windows and RasPiOS workstations, usb-serial and network connections.

### DISCUSSION THREAD

**Discussion by DeeJay** [gusmanb/logicanalyzer#137] [Q&A]
I can see that CLCapture has been available for over 2 years, so I may be misunderstanding how to use it.

Whatever I try, I get a .csv where all rows and all columns (other than the first row of channel names) contain Zero.

Testing with current V5.1 released code.

Tried with basic Pico and the appropriate UF2. Also with a PicoW and both variants of UF2 firmware.

Tried with CLCapture on both RasPiOS (Debian arm64) and Windows10 workstations.

The Pico running the LA firmware is being fed a serial UART message on its Physical pin #15 - LA Channel #14. Running the graphical LA app correctly acquires the input signal. The arguments supplied to CLCcapture are intended to be those that are used to configure the acquisition in ./LogicAnalyzer. 

If I disable the serial signal generator so there is no test signal the LA does not start the capture until I enable to generator source, so I am confident that the Trigger data is being applied to the correct pin (Pin 15, Channel 14) on the Pico.

Where supported by the PicoW firmware I have tried both usb-serial and tcp/ip network connections from the workstation to the Pico.

A typical invocation of CLCapture looks like this -

```
C:\Users\David\Downloads\ClCapture-5.1.0.0-win-x64\win-x64>CLCapture capture COM4 500000  10,11,12,13:Null,14:Serial,15,16,17 512 8192 0 TriggerType:edge,Channel:14,Value:1 capture.csv
Opening logic analyzer in COM4...
Conneced to device LOGIC_ANALYZER_PICO_V5_1 in port/address COM4
Starting edge triggered capture...
Capture running...
Capture complete, writting output file...
Done.

```

(This isn't urgent for me. I don't HAVE to be able to generate a .csv of the data. This is largely curiosity to compare LogicAnalyzer with the two other Sigrok-based Pico LA solutions I am aware of.)

_{edit: added later... }_

It seems to be a regression failure. The capture works as expected with the Pico .UF2 and Win10 CLCapture.exe at Release v2.1.1.0 Jan 2023.)

**Comment by gusmanb** [comment]
I will review it on V6.0, the command line is rarely used so it's possible that is broken.

If you still want to test it with Sigrok you can use the normal app and use the "Export..." option from the menu.

Cheers.

**Reply by DeeJay** [reply]
No pressure for a solution, just a possible insight...

Maybe this is an instance of the "Channels must be sequential with no gaps" feature?

I get captured data in the .csv file if I capture from Channel #1. (The tests I documented above used Channel 14. Only because the bottom left corner of the board is easier to patch reliably with poor eyesight and shaky hands... )

**Reply by gusmanb** [reply]
Don't think so, the problem with the empty channels was exclusive to V6.0 because of the changes that I'm doing to the driver. 

In any case, I'm moving all the code that deals with the capture formats and parsings to the driver to have a single code base shared between the two apps, what is there right now is a bit of a mess, I created the CLCapture as a quick workaround for people to be able to use the analyzer with sigrok and is not precissely "top notch code" 😆, a lot of innecessary duplications and very difficult to mantain. With V6.0 all this goes away, I'm rewritting nearly all the code in the CLCapture.

Cheers.
